src/common/buffer.cc

```diff
@@ -1791,7 +1791,7 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
         if (gap > len) gap = len;
     //cout << "append first char is " << data[0] << ", last char is " << data[len-1] << std::endl;
         append_buffer.append(data, gap);
-        append(append_buffer, append_buffer.end() - gap, gap);	// add segment to the list
+        append(append_buffer, append_buffer.length() - gap, gap);	// add segment to the list
```

Comment:
@YankunLi this issue impacts both hammer and jewel, could you file a ticket on tracker? and put more details in the commit message? then add a test in `src/test/bufferlist.cc`. also, please add a "Fixes:" line at end of your commit message to connect this commit to the tracker ticket.